File: reuse_java/src/test/java/com/kmmania/AthleteImplTest.java
package com.kmmania;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class AthleteImplTest {

    @Test
    @DisplayName("mySpeciality() method should return correct speciality")
    void mySpecialityShouldReturnCorrectSpeciality() {
        // Given
        String speciality = "Running";
        AthleteImpl athlete = new AthleteImpl(speciality);

        // When
        String result = athlete.mySpeciality();

        // Then
        assertThat(result).isEqualTo("My speciality is Running.");
    }
}

File: reuse_kotlin/src/test/kotlin/AthleteImplTest.kt
import com.kmmania.AthleteImpl
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class AthleteImplTest {

    @Test
    fun `mySpeciality() method should return correct speciality`() {
        // Given
        val speciality = "Running"
        val athlete = AthleteImpl(speciality)

        // Then
        val result = athlete.mySpeciality()

        // When
        assertThat(result).isEqualTo("My speciality is Running.")
    }
}